File: 0038-count-and-say/0038-count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
     	if (1== n) return "1";  

    	if (2 == n) return "11";

    	

    	string res="11";

    	string s;

    

    	for (int i = 2; i < n; i++){   

    
    		int len = res.size();

            


    		for (int j = 0; j < len; j++){  

    		    

    		    int count=1;

    

            

				while ((j + 1 < len && res[j] == res[j + 1])){

					count++;    

					

				
j++;}
                

           

    			s+=to_string(count) + res[j];

    		}

    

          

    		res = s;

    		

    		

    		

    
s.clear();
    	}

    

    	return res;

    }

};

File: 0038-count-and-say/0038-count-and-say.dart
class Solution {
  String countAndSay(int n) {
  String say(String s) {
    String res = "";
    int i = 0;
    while (i < s.length) {
      int count = 1;
      while (i + 1 < s.length && s[i] == s[i + 1]) {
        count++;
        i++;
      }
      res += "$count${s[i]}";
      i++;
    }
    return res;
  }

  String s = "1";
  for (int i = 1; i < n; i++) {
    s = say(s);
  }
  return s;
}

}